### DISCUSSION TITLE
Support for the new Ring Car Cam

### DISCUSSION THREAD

**Discussion by MuscleGeek69** [tsightler/ring-mqtt#580] [Feature Requests]
Hello,

I would like to know if there are plans to support the new ring car cam, especially the GPS feature. 

Thanks in advance.

**Comment by tsightler** [comment]
Ring car cam is a completely different device from existing Ring cameras and appears to use a different API and also appears to only supprot HEVC video encoding.  The ring-client-api project has not added support for car cam and thus it has no real chance of being added here until that is done.  Most likely someone with a car cam is going to have to reverse engineer the API endpoints so no telling when that will happen.

I'm actually interested in a car cam but, unfortunately, it's not compatible with my vehicle, so it probably won't be me that does the reverse engineering work.  If support is eventually added to ring-client-api, then I'd be willing to add it to this project.

**Reply by alvinchen1** [reply]
Yes, wouldn't hurt to add a discussion to https://github.com/dgreif/ring/discussions.  Dev is super cool and has quickly turned around other models (like when I upgraded from Spotlight Cam to Spotlight Pro, etc) that I got in the past.  But because as @tsightler said, this is a different API, might take a bit longer. Doesn't hurt to ask!

**Reply by tsightler** [reply]
Well, adding a new model camera vs what appears to be a completely new device is pretty different.  There is already a feature request for Car Cam support on ring-client-api issues page, but the reality is that, until someone that has a Car Cam does the reverse engineering required, or at least is willing to provide access to one of us willing to do the work on that, then support is not likely to happen.

My hope is that, overall, it would be pretty basic.  I'm thinking the events API might be the same, which would make viewing recordings and streaming overall the same, and that it just uses a push API, similar to the intercom, for events, since that seems to be the new model Ring is using for those types of devices.  But all of that is just guesswork until we have access to one.

**Reply by alvinchen1** [reply]
Yup!  You do enough already!  Let someone else do a little heavy lifting... Thanks for everything!

**Reply by MuscleGeek69** [reply]
Thank you for the explanation. I would like to thank you @tsightler and everyone involved in this project and the whole HA ecosystem. Thanks to all of you I run an autonomous house and never have to worry about talking to Alexa :D
